rules/license.bzl

```diff
@@ -80,51 +61,66 @@ _license = rule(
+            doc = "A human readable name used to organize licenses into categories." +
+                  " This is used in google3 to differentiate third party licenses used" +
+                  " for compliance versus internal licenses used by SLAsan for internal" +
+                  " teams' SLAs.",
+        ),
     },
 )
 
 # buildifier: disable=function-docstring-args
-def license(name,
-            copyright_notice = None,
-            license_kinds = None,
-            license_text = None,
-            package_name = None,
-            package_url = None,
-            package_version = None,
-            tags = None,
-            **kwargs):
+def license(
+        name,
+        license_text = "LICENSE",
+        visibility = ["//visibility:public"],
+        license_kind = None,
+        license_kinds = None,
+        copyright_notice = None,
+        package_name = None,
+        package_url = None,
+        package_version = None,
+        namespace = "compliance",
+        tags = []):
     """Wrapper for license rule.
 
     Args:
       name: str target name.
+      license_text: str Filename of the license file
+      visibility: list(label) visibility spec
+      license_kind: label a single license_kind. Only one of license_kind or license_kinds may
+                    be specified
       license_kinds: list(label) list of license_kind targets.
-      license_kind: label a single license_kind. Only one of license_kind or
-                    license_kinds may be specified
       copyright_notice: str Copyright notice associated with this package.
-      package_name: str A human readable name identifying this package. This
-                    may be used to produce an index of OSS packages used by
-                    an applicatation.
-      package_url: The URL this instance was downloaded from.
-      package_version: The version number of this package. This should be a
-                       value that increases over time, rather than a commit
-                       hash.
-      kwargs: Other things may be specified, but they are explicitly ignored.
+      package_name : str A human readable name identifying this package. This
+                     may be used to produce an index of OSS packages used by
+                     an application.
+      tags: list(str) tags applied to the rule
     """
-    single_kind = kwargs.pop("license_kind", default = None)
-    if single_kind:
+    if license_kind:
         if license_kinds:
             fail("Can not use both license_kind and license_kinds")
-        license_kinds = [single_kind]
-    tags = tags or []
+        license_kinds = [license_kind]
+
+    # Make sure the file exists as named in the rule. A glob expression that
+    # expands to the name of the file is not acceptable.
+    srcs = native.glob([license_text])
+    if len(srcs) != 1 or srcs[0] != license_text:
+        fail("Specified license file doesn't exist: %s" % license_text)
+
```

Comment:
@aiuto `native.glob` doesn’t support `@//` as reported [here](https://github.com/bazelbuild/bazel/issues/1346), so when we generate a license defined like:
```
license(
	name = "javax-mail-LICENSE",
	license_text = "@my_external_compliance_repo//path:LICENSE-javax-mail-1.4.7",
...
```
and run a policy check, the check crashes.

Could we remove this code or execute a different way to allow for loading license files from an external repository?

Reply:
I believe this issue is representing this problem: https://github.com/bazelbuild/rules_license/issues/68